dace/frontend/fortran/fortran_parser.py

```diff
@@ -2235,9 +1444,324 @@ def subroutine2sdfg(self, node: ast_internal_classes.Subroutine_Subprogram_Node,
+                                            array.dtype,
+                                            storage=array.storage,
+                                            strides=strides,
+                                            offset=offsets_zero)
+        
+        memlet=Memlet.simple(concatenated_name + "_" + ast_utils.get_name(member) + "_" + str(
+                            self.struct_view_count), subset)
+
+        return self.add_accesses_and_edges(sdfg,view_name,view, array, substate, last_read, last_written, read, write,memlet)
+
+
+    def add_array_to_element_view_pair_in_tower(self, sdfg: SDFG, array: dat.Array, name_chain: List[str], member: ast_internal_classes.FNode, substate: SDFGState, last_read: nd.AccessNode, last_written: nd.AccessNode, read: bool, write: bool,subset):
+        
+        stype=array.stype
+        view_to_member = dat.View.view(stype)
+        concatenated_name = "_".join(name_chain)
+        view_name=concatenated_name + "_" + ast_utils.get_name(member) + "_m_" + str(
+                            self.struct_view_count)
+        memlet=Memlet.simple(concatenated_name + "_" + ast_utils.get_name(member) + "_" + str(
+                            self.struct_view_count), subset)
+
+        return self.add_accesses_and_edges(sdfg,view_name,view_to_member, array, substate, last_read, last_written, read, write,memlet)
+
+    def add_basic_view_pair_in_tower(self, sdfg: SDFG, array: dat.Array, name_chain: List[str], member: ast_internal_classes.FNode, substate: SDFGState, last_read: nd.AccessNode, last_written: nd.AccessNode, read: bool, write: bool):
+        view_to_member = dat.View.view(array)
+        concatenated_name = "_".join(name_chain)
+        view_name=concatenated_name + "_" + ast_utils.get_name(member) + "_" + str(
+                            self.struct_view_count)
+        memlet=Memlet.from_array(concatenated_name + "." + ast_utils.get_name(member), array)
+        return self.add_accesses_and_edges(sdfg,view_name,view_to_member, array, substate, last_read, last_written, read, write,memlet)
+        
+
+    def add_accesses_and_edges(self,sdfg: SDFG,view_name:str,view_to_member:dat.View, array: dat.Array, substate: SDFGState, last_read: nd.AccessNode, last_written: nd.AccessNode, read: bool, write: bool,memlet:Memlet):
+        sdfg.arrays[view_name] = view_to_member
+        if read:
+            new_read=substate.add_read(view_name)
+            substate.add_edge(last_read, None, new_read, None, dpcp(memlet))
+            last_read=new_read
+        if write:
+            new_written=substate.add_write(view_name)
+            substate.add_edge( new_written, None,last_written, None, dpcp(memlet))
+            last_written=new_written
+        
+        return last_read, last_written    
+
+    def process_variable_call(self, variable_in_calling_context: ast_internal_classes.FNode, local_name:ast_internal_classes.FNode,  sdfg: SDFG, new_sdfg: SDFG, substate:SDFGState, read:bool,write:bool):
```

Comment:
What does this return? As far as I can tell, currently it returns a `Tuple[bool, List[??]]`. But the second component is there only if the first component is `True`. Does it actually intend to return an `Optional[List[??]]`?

Even then, why a `List[??]`? As far as I can tell, currently it always returns a 4-item list (when it's not `None`). Maybe it actually is a `Tuple[?, ?, ?, ?]` (tuples are easier to reason about due to their fixed lengths)?

Finally, at the very end of the function, it doesn't return anything (on the final branch, or at the end in general). This effectively makes it return a `None`, which is completely different than the rest of the current returns.